TSLint: jsx-no-lambda

mgolawski opened the pull request:
--Get rid of `jsx-no-lambda` rules violation.
--Set printWidth to `120` for prettier
--Change some stateless components to function components

bklukaczewski commented on src/components/Button.tsx:
```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, { SFC } from 'react';
```
You imported SCF but later use `React.SFC` ;)

bklukaczewski commented on src/screens/updatePlan/CreatePlanItemButton.tsx:
```diff
@@ -59,3 +18,46 @@ const styles = StyleSheet.create({
   },
 });
+
+export const CreatePlanItemButton: SFC<Props> = (props: Props) => {
```
What is rationale behind this change?
- Functional components does not support hot reloading
- Pure components are better for performance, as they do only shallow prop comparison (might not apply to this particular component)

What do we gain in return?

mgolawski replied:
Since I have mostly standard React background, I am not familiar with hot reloading. What about that https://github.com/bvic23/babel-plugin-functional-hmr? Maybe it can fix the problem with functional components?

What do we gain in return?
- We can use `useCallback` hook for memoizing the callback method, and avoid the anonymous function declaration in every component re-render. This is a performance bottleneck.
- We get rid of useless `this` keyword for this component. 
- In this particular scenario shallow comparison does not give us any benefits.

mgolawski replied:
It seems that hooks, sfc, and hot reloading don't like each other. I will dig into that deeper tomorrow, since my knowledge on that topic is negligible.

mgolawski commented:
@bklukaczewski I addressed your code review comments, and pushed proper changes. It seems that hot reload for hooks and functional components will be handled in `0.61` version of `react-native` according to the Abramov comment https://github.com/facebook/react-native/issues/18899#issuecomment-506881769.

I also did a merge with your recent changes, so please check if everything is merged correctly. Due to the lack of tests I can't validate if everything went fine.

Thanks.
